python_modules/dagster/dagster/_core/definitions/declarative_automation/automation_condition_tester.py

```diff
@@ -142,13 +143,14 @@ def my_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
         cursor=cursor,
     )
     results, requested_subsets = evaluator.evaluate()
-    new_cursor = cursor.with_updates(
-        evaluation_timestamp=(evaluation_time or datetime.datetime.now()).timestamp(),
-        newly_observe_requested_asset_keys=[],
-        evaluation_id=cursor.evaluation_id + 1,
-        condition_cursors=[result.get_new_cursor() for result in results],
-        asset_graph=asset_graph,
-    )
+    with partition_loading_context(effective_dt=evaluation_time, dynamic_partitions_store=instance):
```

Comment:
should we put this even higher / around the evaluate call too? any downside?

Reply:
no real downsides, but it's kinda nice to put it down here to sorta force ourselves to do the right thing in other codepaths (this is just test code, after all)